### PR TITLE
Protect routes and redirect

### LINKED ISSUE #3: Add Auth0 to project
We need to implement Auth0 in the project.
- If user is not authenticated, show login screen/landing page
- All other routes need to be behind the authentication

### PR DESCRIPTION
**Pull request by pbarrowbyu121**
Added protected routes, rearranged auth0 sign in and sign out to main menu, and it should redirect to app after sign in. Lists and profile are the only protected links right now.

### CODE REVIEW

File: src/auth/index.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      /** Logs the user out and removes their session on the authorization server */
+      logout(o) {
+        return this.auth0Client.logout(o);
+      }
+    },
+    /** Use this lifecycle method to instantiate the SDK client */
+    async created() {
+      // Create a new instance of the SDK client using members of the given options object
+      console.log("REDIReCT URI", redirectUri);
+      console.log("OPTIONS", options);
```
Comment (briancbarrow) [inline src/auth/index.js:94]: Let's remove these console logs

File: src/auth/index.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      this.auth0Client = await createAuth0Client({
+        ...options,
+        client_id: options.clientId,
+        redirect_uri: redirectUri
+      });
+      console.log("AUTH0 CLIENT", this.auth0Client);
```
Comment (briancbarrow) [inline src/auth/index.js:100]: Same with this one

File: src/pages/SignIn.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<template>
+  <q-page
+    class="bg-light-green window-height window-width row justify-center items-center"
+  >
+    <div class="column">
+      <div class="row">
+        <h5 class="text-h5 text-white q-my-md">Booklist App</h5>
+      </div>
+      <div class="row">
+        <q-card square bordered class="q-pa-lg shadow-1">
```
Comment (briancbarrow) [inline src/pages/SignIn.vue:10]: Let's delete this file

File: src/pages/SignIn222.vue [hunk 4]
```diff
@@ -0,0 +1,61 @@
+<template>
```
Comment (briancbarrow) [inline src/pages/SignIn222.vue:1]: Let's remove this

**Review by briancbarrow — APPROVED** [review]
Just address the changes and then you can merge

File: src/auth/index.js (hunk 2 above)
Comment (pbarrowbyu121) [reply src/auth/index.js:100]: Removed

File: src/pages/SignIn.vue (hunk 3 above)
Comment (pbarrowbyu121) [reply src/pages/SignIn.vue:10]: Deleted file

File: src/pages/SignIn222.vue (hunk 4 above)
Comment (pbarrowbyu121) [reply src/pages/SignIn222.vue:1]: File deleted

**Review by SamBrieden — APPROVED** [review]

**Comment by AlexanderPuhl** [comment]
I'm getting the error code `"TypeError: Cannot read property 'loginWithRedirect' of null"`

Unable to login, can we deal with this error and then push up the changes?

**Pull request merged**